src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -105,11 +110,15 @@
     private final List<String> categoryFields;
     private User user;
     private String detectorType;
+    private String resultIndex;
 
     // TODO: support backward compatibility, will remove in future
     @Deprecated
     private DetectionDateRange detectionDateRange;
 
+    public static final int MAX_RESULT_INDEX_NAME_SIZE = 255;
+    public static final String RESULT_INDEX_NAME_PATTERN = "^[a-z0-9_-]*$";
```

Comment:
Does this pattern mean a result index name should start with a-z, 0-9, or _ or -? If so, according to https://discuss.elastic.co/t/index-name-type-name-and-field-name-rules/133039, an index name cannot start with -, _,.

Reply:
Actually I may forget to change the index mapping before pushing commit last night, I remember it should be "[a-z0-9_-]+"

Hard to find out exact pattern for OpenSearch index. We always add CUSTOM_RESULT_INDEX_PREFIX "opensearch-ad-plugin-result-" to custom index mapping.  This pattern here is to do some preliminary check, a complete index name check will be done by OpenSearch create index request if this name regex matches.